mono/utils/mono-state.c

```diff
@@ -145,18 +145,43 @@ static const char *configured_timeline_dir;
 static MonoSummaryTimeline log;
 
 static void
-file_for_summary_stage (const char *directory, MonoSummaryStage stage, gchar *buff, size_t sizeof_buff)
+file_for_stage_breadcrumb (const char *directory, MonoSummaryStage stage, gchar *buff, size_t sizeof_buff)
 {
 	g_snprintf (buff, sizeof_buff, "%s%scrash_stage_%d", directory, G_DIR_SEPARATOR_S, stage);
 }
 
 static void
-create_stage_mark_file (void)
+file_for_dump_reason_breadcrumb (const char *directory, const char *dump_reason, gchar *buff, size_t sizeof_buff)
+{
+	g_snprintf (buff, sizeof_buff, "%s%scrash_reason_%s", directory, G_DIR_SEPARATOR_S, dump_reason);
```

Comment:
Btw both of these breadcrumbs should be `dump_stage` and `dump_reason` (since we now have `crash_stage` breadcrumbs from managed exception reporting)

Reply:
I think colloquially an unhandled managed exception is a "crash" too. I wouldn't worry too much about the distinction in the file names.